flareon-codegen/src/model.rs

```diff
@@ -80,58 +90,125 @@ impl ModelOpts {
             original_name.to_string().to_case(Case::Snake)
         };
 
+        let primary_key_field = self.get_primary_key_field(&fields)?;
+
         Ok(Model {
             name: self.ident.clone(),
             original_name,
             model_type: args.model_type,
             table_name,
+            pk_field: primary_key_field.clone(),
             fields,
         })
     }
+
+    fn get_primary_key_field<'a>(&self, fields: &'a [Field]) -> Result<&'a Field, syn::Error> {
+        let pks: Vec<_> = fields.iter().filter(|field| field.primary_key).collect();
+        if pks.is_empty() {
+            return Err(syn::Error::new(
+                self.ident.span(),
+                "models must have a primary key field, either named `id` \
+                or annotated with the `#[model(primary_key)]` attribute",
+            ));
+        }
+        if pks.len() > 1 {
+            return Err(syn::Error::new(
+                pks[1].field_name.span(),
+                "composite primary keys are not supported; only one primary key field is allowed",
+            ));
+        }
+
+        Ok(pks[0])
+    }
 }
 
 #[derive(Debug, Clone, FromField)]
 #[darling(attributes(model))]
 pub struct FieldOpts {
     pub ident: Option<syn::Ident>,
     pub ty: syn::Type,
+    pub primary_key: darling::util::Flag,
     pub unique: darling::util::Flag,
 }
 
 impl FieldOpts {
+    #[must_use]
+    fn find_type(&self, type_to_check: &str, symbol_resolver: &SymbolResolver) -> bool {
```

Comment:
I'll go with something slightly different, `has_type`. `is_type` suggests that the type being looked for is the only one (while it might be a composition of multiple generic types), while `type_exists` is somewhat vague.